src/coreclr/nativeaot/Runtime/gcheaputilities.cpp

```diff
@@ -47,6 +49,425 @@ extern "C" HRESULT GC_Initialize(
 
 #ifndef DACCESS_COMPILE
 
+enum GC_LOAD_STATUS {
+    GC_LOAD_STATUS_BEFORE_START,
+    GC_LOAD_STATUS_START,
+    GC_LOAD_STATUS_DONE_LOAD,
+    GC_LOAD_STATUS_GET_VERSIONINFO,
+    GC_LOAD_STATUS_CALL_VERSIONINFO,
+    GC_LOAD_STATUS_DONE_VERSION_CHECK,
+    GC_LOAD_STATUS_GET_INITIALIZE,
+    GC_LOAD_STATUS_LOAD_COMPLETE
+};
+
+// Load status of the GC. If GC loading fails, the value of this
+// global indicates where the failure occurred.
+GC_LOAD_STATUS g_gc_load_status = GC_LOAD_STATUS_BEFORE_START;
+
+// The version of the GC that we have loaded.
+VersionInfo g_gc_version_info;
+
+class MyGCToEEInterface : public IGCToCLR {
```

Comment:
Why do we need to add all this to the shared file? Should this be in the clrgc.enabled.cpp file only?

Reply:
Can we put a whole class in a different lib/a file? I was hoping that the linker is smart enough so that if there is no call site to construct this, it won't get into the binary.

Reply:
It would make it much easier to see all the optional code and ensure that it does not get pulled in by accident.

It would match how the optional support for eventpipe is structured. We should use the same approach for all optional native AOT components.